docs/style_guide.adoc

```diff
@@ -96,79 +94,24 @@ g_environment: someval
 ----
 
 === Ansible Role Variables
-Ansible role variables are defined as variables contained in (or passed into) a role.
-
 '''
-[[Role-variables-MUST-have-a-prefix-of-atleast-3-characters-See.below.for.specific.naming.rules]]
+[[Ansible-role-variable-prefix]]
 [cols="2v,v"]
 |===
-| <<Role-variables-MUST-have-a-prefix-of-atleast-3-characters-See.below.for.specific.naming.rules, Rule>>
-| Role variables MUST have a prefix of at least 3 characters. See below for specific naming rules.
+| <<Ansible-role-variable-prefix, Rule>>
+| Role variables MUST have a prefix of `r_<role_name>_`
 |===
 
-==== Role with 3 (or more) words in the name
+.Definition
+Ansible variables contained in or passed into a role.
 
-Take the first letter of each of the words.
-
-.3 word example:
-* Role name: made_up_role
-* Prefix: mur
-[source]
-----
-mur_var1: value_one
-----
+.Purpose
+Name variables to avoid name collision between roles or playbooks.
```

Comment:
I hit very similar situation when refactoring etcd roles:

etcd_common/default/main.yml
```yaml
r_etcd_common_service: "{{ 'etcd' if r_etcd_common_runtime == 'host' else 'etcd_service' }}"
```

etcd/tasks/main.yml
```yaml
- name: Restart etcd
  service:
    name: "{{ r_etcd_common_service }}"
    state: restarted
```

Then, when calling role `etcd` one needs to set `r_etcd_common_runtime`. One can either:
```yaml
- roles:
    - role: etcd
      r_etcd_common_runtime: "docker"
```
in which case we must be aware the `etcd` depends on `etcd_common` (but that can change over time). Or we can have `etcd` role define `r_etcd_runtime` varible and have `etcd` set `r_etcd_common_runtime`:
```
```yaml
- roles:
    - role: etcd
      r_etcd_runtime: "docker"
```

The latter has advantage of us not knowing about etcd dependencies. If `etcd_common` gets renamend to `etcd_dog`, API of `etcd` will remain untouched.